akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -40,7 +46,13 @@ sealed abstract class OutPort { self: Outlet[_] ⇒
   /**
    * INTERNAL API
    */
-  private[stream] var id: Int = -1
+  @volatile private[stream] var id: Int = -1
+
+  /**
+   * INTERNAL API
+   */
+  @volatile private[stream] var mappedTo: OutPort = this
```

Comment:
It was not a bug. The ID was only used by the interpreter before, and it set its value inside an actor before starting. Multiple initialization could have happened, but that is not an issue. Might not be needed even now, but I didn't want to risk it.